### PR TITLE
Add basic linkify-matrix tests

### PR DESCRIPTION
**Pull request by Palid**
Add some basic linkify-matrix tests before upgrading the dependencies to a newer version.

---
This PR currently has no changelog labels, so will not be included in changelogs.

A reviewer can add one of: `T-Deprecation`, `T-Enhancement`, `T-Defect`, `T-Task` to indicate what type of change this is, or add `Type: [enhancement/defect/task]` to the description and I'll add them for you.

Preview: https://619ff61d2458c7374904f648--matrix-react-sdk.netlify.app
⚠️ Do you trust the author of this PR? Maybe this build will steal your keys or give you malware. Exercise caution. Use test accounts.

### CODE REVIEW

File: test/linkify-matrix-test.ts [hunk 1]
```diff
@@ -0,0 +1,227 @@
+import * as linkify from "linkifyjs";
```
Comment (t3chguy) [inline test/linkify-matrix-test.ts:1]: needs copyright header

File: test/linkify-matrix-test.ts [hunk 2]
```diff
@@ -0,0 +1,227 @@
...
+import linkifyMatrix from '../src/linkify-matrix';
+
+describe('linkify-matrix', () => {
+    beforeAll(() => {
+        linkifyMatrix(linkify);
```
Comment (t3chguy) [inline test/linkify-matrix-test.ts:6]: normally we do this right after the import, we should probably do that to match all other callers

File: test/linkify-matrix-test.ts [hunk 3]
```diff
@@ -0,0 +1,227 @@
...
+            }]));
+        });
+        it('accept #foo:bar.com', () => {
+            const test = '#foo:bar.com';
+            const found = linkify.find(test);
+            expect(found).toEqual(([{
+                href: "#foo:bar.com",
+                type: "roomalias",
+                value: "#foo:bar.com",
+
```
Comment (t3chguy) [inline test/linkify-matrix-test.ts:34]: stray NL

File: test/linkify-matrix-test.ts [hunk 4]
```diff
@@ -0,0 +1,227 @@
...
+            },
+            {
+                href: "http://bar.com",
+                type: "url",
+                value: "bar.com",
+            }]));
+        });
+        // This should not parse correctly, but it's been working this way in the previous version too
+        it.skip('does not parse multiple room aliases in one string', () => {
+            const test = '#foo:bar.com-baz.com';
```
Comment (t3chguy) [inline test/linkify-matrix-test.ts:111]: How is `#foo:bar.com-baz.com` multiple room aliases? `bar.com-baz.com` is a valid domain
subdomain=`bar`
domain=`com-baz`
tld=`com`

File: test/linkify-matrix-test.ts [hunk 5]
```diff
@@ -0,0 +1,227 @@
...
+        // It does not parse a single user ID without domain
+        it.skip('properly parses @foo', () => {
+            const test = "@foo";
+            const found = linkify.find(test);
+            expect(found).toEqual(([{
+                href: "@foo",
+                type: "userid",
+                value: "@foo",
+            }]));
+        });
```
Comment (t3chguy) [inline test/linkify-matrix-test.ts:182]: why do we have a test which we know is invalid and skip it?

File: test/linkify-matrix-test.ts [hunk 6]
```diff
@@ -0,0 +1,227 @@
...
+                href: "#foo",
+                type: 'roomalias',
+                value: '#foo',
+            },
+            {
+                href: "http://bar.com",
+                type: "url",
+                value: "bar.com",
+            }]));
+        });
```
Comment (t3chguy) [inline test/linkify-matrix-test.ts:108]: I would expect this to not parse the room though, `#foo` is not a valid roomalias.
This test is invalid.

**Review by t3chguy — CHANGES_REQUESTED** [review]

File: test/linkify-matrix-test.ts (hunk 3 above)
Comment (Palid) [reply test/linkify-matrix-test.ts:34]: Fixed, but please fix linter rules instead... 😡

File: test/linkify-matrix-test.ts (hunk 6 above)
Comment (Palid) [reply test/linkify-matrix-test.ts:108]: Fixed the test to make it valid instead

File: test/linkify-matrix-test.ts (hunk 4 above)
Comment (Palid) [reply test/linkify-matrix-test.ts:111]: Must have confused it with something else, there's been a lot of tests I did in the meantime and probably did some mess within rebases and commits. Fixed, thanks for catching!

File: test/linkify-matrix-test.ts [hunk 7]
```diff
@@ -0,0 +1,238 @@
...
+        it('does not parse room alias with too many separators', () => {
+            const test = '#foo:::bar.com';
+            const found = linkify.find(test);
+            expect(found).toEqual(([{
+                href: "http://bar.com",
+                type: "url",
+                value: "bar.com",
+            }]));
+        });
+        it('does not parse multiple room aliases in one string', () => {
```
Comment (t3chguy) [inline test/linkify-matrix-test.ts:118]: this test name doesn't seem to match the fixture

**Review by t3chguy — APPROVED** [review]
LGTM otherwise

**Comment by t3chguy** [comment]
Hmm, I didn't see that auto-merge was enabled. @Palid could you include a fix for the test name in the PR for upgrading linkify whenever you do it please

**Comment by Palid** [comment]
> Hmm, I didn't see that auto-merge was enabled. @Palid could you include a fix for the test name in the PR for upgrading linkify whenever you do it please

Will do as a follow-up, the linkify upgrade may take a tiny bit longer due to some parser changes. 😅

**Pull request merged**